master/networking/ip-autodetection.md

```diff
@@ -0,0 +1,195 @@
+    ipv4IPIPTunnelAddr: 192.168.0.1
+```
+
+#### Auto detection methods 
+
+By default, {{site.prodname}} uses the **first-found** method; the first valid IP address on the first interface (excluding local interfaces such as the docker bridge). However, you can change the default method to any of the following:
+
+- IP addresses or domains (**can-reach**)
+- regex to include matching interfaces (**interface**)
+- regex to exclude matching interfaces (**skip-interface**)
+
+For details on auto detection methods, see [calicoctl node run]({{site.baseurl}}/{{page.version}}/reference/calicoctl/node/run#displaying-the-help-text-for-calicoctl-node-run-command).
+
+#### Manually configure IP address and subnet
+
+There are two ways to manually configure an IP address and subnet:
+
+- {{site.prodname}} node container (start/restart)   
+  Use environment variables and command line options to set values for nodes. 
+
+- {{site.prodname}} node resource   
+  Update the node resource.
+
+##### Using environment variables and node resource
+
+Because you can configure IP address and subnet using either environment variables or node resource, the following table describes how values are synchronized. 
+
+| **If this environment variable...** | **Is...**                                             | **Then...**                                                  |
+| ----------------------------------- | ----------------------------------------------------- | ------------------------------------------------------------ |
+| IP/IP6                              | Explicitly set                                        | The specified values are used, and the Node resource is updated. |
+|                                     | Set to autodetect                                     | The requested method is used (first-found, can-reach, interface, skip-interface), and the Node resource is updated. |
+|                                     | Not set, but Node resource has IP/IP6 values          | Node resource value is used.                                 |
+| IP                                  | Not set, and there is no IP value in Node resource    | Auto detects an IPv4 address and subnet, and updates Node resource. |
+| IP6                                 | Not set, and there is a notIP6 value in Node resource | No IP6 routing is performed on the node.                     |
+
+### How to
+
+- [Change the auto detection method](#change-the-auto-detection-method)
+- [Manually configure IP address and subnet for a node](#manually-configure-ip-address-and-subnet-for-a-node)
+
+#### Change the auto detection method
+
+As noted previously, the default auto detection method is **first valid interface found** (first-found). To use a different auto detection method, use the following {{site.prodnamedash}} node run command, specifying the method:
+
+- **IPv4**
+
+  ```
+  calicoctl node run  --ip-autodetection-method=<autodetection-method>
```

Comment:
So, calicoctl node run is not really used in Kubernetes systems and I think is probably just outdated at this point. Some people might still use it, but it's certainly not the mainline by a long shot.

I suspect we should document a Kubernetes-appropriate way of setting these (namely, using environment variables) and ditch the `calicoctl node run` stuff altogether

Reply:
Maybe this would be a good "tweener" task for Laurence?

Reply:
yep. go ahead.

Reply:
c03c741171c2b2291e840186504fbfc998304b1f

Reply:
Waiting for Laurence to add kubectl commands.